src/units/abilities.cpp

```diff
@@ -1168,52 +1168,137 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 	}
 }
 
-//Functions used for change damage_type list with damage
-static std::optional<std::string> select_damage_type(const unit_ability_list& abil_list, const std::string& type)
+//functions for select damage type who must be returned for each attribute used in damage_type()
+static std::string select_replacement_type(const unit_ability_list& damage_type_list)
 {
+	std::unordered_map<std::string, unsigned int> type_count;
+	unsigned int max = 0;
+	for(auto& i : damage_type_list) {
+		const config& c = *i.ability_cfg;
+		if(!c["replacement_type"].empty()) {
```

Comment:
Is there a reason why this does not use has_attribute? Is this a joke?

Reply:
i forget to change that, sorry.

Reply:
fixed